### ISSUE TITLE
Fetch `interface type` from `ietf` interface yang model

### ISSUE THREAD

**Issue by frederic-loui** [holo-routing/holo#22]
Fetch `interface type` from `ietf` interface yang model

For now when we configure `holo` via `holo-cli`, `interface type` is mandatory and entered in a free form.
However, `type` seems to be a value from `ietf-interfaces` or `ietf-if-extensions` yang model 

It would be ideal to be able to select interface type from a discrete list built from `ietf-interfaces` or `ietf-if-extensions` or other yang model

This will enforce interface configuration consistency

**Comment by rwestphal** [comment]
Absolutely. It'd be awesome if we could use the TAB key to choose from different interface types.

This could be extended to all YANG leaves whose type is `identityref` (e.g. control-plane protocol types).

As a first step, we should investigate if libyang supports iterating over all identities of a specified type. Once confirmed, we can use this capability to generate auto-completion suggestions.

I'll transfer this issue to the *holo-cli* repo.
